src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Macros/RegexMatchMacro.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Text.RegularExpressions;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Core.Contracts;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros.Config;
+using Microsoft.TemplateEngine.Utils;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros
+{
+    public class RegexMatchMacro : IDeferredMacro
+    {
+        public Guid Id => new Guid("AA5957B0-07B1-4B68-847F-83713973E86F");
+
+        public string Type => "regexMatch";
+
+        public void EvaluateConfig(IEngineEnvironmentSettings environmentSettings, IVariableCollection vars, IMacroConfig rawConfig, IParameterSet parameters, ParameterSetter setter)
+        {
+            string value;
+
+            if (!(rawConfig is RegexMatchMacroConfig config))
+            {
+                throw new InvalidCastException("Couldn't cast the rawConfig as RegexMatchMacroConfig");
+            }
+
+            if (!vars.TryGetValue(config.SourceVariable, out object working))
+            {
+                value = parameters.TryGetRuntimeValue(environmentSettings, config.SourceVariable, out object resolvedValue, true)
+                    ? resolvedValue.ToString()
+                    : string.Empty;
+            }
+            else
+            {
+                value = working?.ToString() ?? "";
+            }
+
+            bool result = Regex.IsMatch(value, config.Pattern);
```

Comment:
Do we want to try-catch for malformed regex'es? (There are other places outside this review where we should consider doing the same).